src/tuning/kernels/xgemm.cpp

```diff
@@ -190,10 +226,39 @@ void StartVariation(int argc, char *argv[]) {
   }
 }
 
+// Test multiple heuristics if kArgMultiSearchStrategy is enabled
+// Otherwise, run the heuristic specified in kArgMultiSearchStrategy
+void TestHeuristic(int argc, char *argv[]){
+
+  auto command_line_args = clblast::RetrieveCommandLineArguments(argc, argv);
+  auto help = std::string{""};
+  auto heuristic_selected = clblast::GetArgument(command_line_args, help, clblast::kArgMultiSearchStrategy, 0); 
+  auto multi_search_strategy = clblast::GetArgument(command_line_args, help, clblast::kArgMultiSearchStrategy, 0);
+
+  if(multi_search_strategy){
+    StartVariation<1>(argc, argv);
+    StartVariation<2>(argc, argv);
+  }
```

Comment:
This multi-search-strategy option: is that really needed? It complicates the code quite a lot and doesn't add much for the end user I would say. The end-user will just want to run the tuner as it is best to do it (e.g. using PSO if that's the best strategy) and don't care otherwise. For those that do experiments on the search strategy and other advanced users: they can now use the command-line option to switch between search strategies, right? I don't really see the use-case for doing two strategies at once.